File: KU01/13/thief.cpp
#include<iostream>
#include <set>
#include <time.h>

using namespace std;

int main() {

    int x, y, z, index;
    cin >> x >> y >> z;
    set<int, greater<int> > ans;
    
    int i = 1;
    ans.insert(1);

    clock_t start = clock(); // Start measuring time

    while (index != z && index != 1)
    {
        i = i + y;
        if (i > x)
        {
            i = i - x;
        }
        ans.insert(i);
        // cout << "i = " << i;
        index = i;
    }
    cout << ans.size() << endl;

    // Stop measuring time and calculate the elapsed time
    clock_t end = clock();

    double elapsed = double(end - start)/CLOCKS_PER_SEC;
    
    printf("Time measured: %.40f seconds.\n", elapsed);
    
    return 0;
}

File: KU01/13/thief.py
import time

n, k, t = input().split()

t = int(t)
round_count = 1
current = 1
final_set = set()
final_set.add(1)
result = 0
j = 0
start = time.time()
while True:
    # print("current = ", current)
    cal = current + int(k)
    if cal > int(n):
        cal = cal - int(n)
    # print(cal)
    final_set.add(cal)
    current = cal
    # print("cal = ", cal)
    if cal == 1 or cal == t:
        break

print(len(final_set))

end = time.time()
print(f'Time measured: {end - start} seconds.')
